backend/scoreserver/domain/notice.go

```diff
@@ -0,0 +1,171 @@
+package domain
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/gofrs/uuid/v5"
+)
+
+type Notice notice
+type notice struct {
+	id             uuid.UUID
+	path           string
+	title          string
+	markdown       string
+	EffectiveFrom  *time.Time
+	EffectiveUntil *time.Time
+}
+
+type NoticeData struct {
+	ID             uuid.UUID
+	Path           string
+	Title          string
+	Markdown       string
+	EffectiveFrom  *time.Time
+	EffectiveUntil *time.Time
+}
+
+func (n *Notice) Path() string {
+	return n.path
+}
+
+func (n *Notice) Title() string {
+	return n.title
+}
+
+func (n *Notice) Markdown() string {
+	return n.markdown
+}
+
+func ListNotices(ctx context.Context, eff NoticeReader) ([]*Notice, error) {
+	data, err := eff.ListNotices(ctx)
+	if err != nil {
+		return nil, WrapAsInternal(err, "failed to get descriptive notices")
+	}
+	notices := make([]*Notice, 0, len(data))
+	for _, d := range data {
+		notice := &Notice{
+			id:             d.ID,
+			path:           d.Path,
+			title:          d.Title,
+			markdown:       d.Markdown,
+			EffectiveFrom:  d.EffectiveFrom,
+			EffectiveUntil: d.EffectiveUntil,
+		}
+		notices = append(notices, notice)
+	}
+	return notices, nil
+}
+
+type NoticeRawData struct {
+	PageID   string
+	PagePath string
+	Content  string
+	Title    *string
+}
+
+func SaveNotice(ctx context.Context, eff NoticeWriter, notice *NoticeData) error {
```

Comment:
ドメインを利用する側に *Data 系の型(DTO)が露出するのは避けたい。ドメインオブジェクト(Notice etc...)は制約とロジックを内側に持つが、これらの型は単純にデータを運ぶだけであり制約の表現力に欠けている

```suggestion
func (n *Notice) Save(ctx context.Context, eff NoticeWriter) error {
```